var/spack/repos/builtin/packages/openspeedshop/package.py

```diff
@@ -96,7 +96,7 @@ class Openspeedshop(CMakePackage):
 
     depends_on("libxml2")
 
-    depends_on("qt@3", when='gui=qt3')
+    depends_on("qt@3.3.8b", when='gui=qt3')
```

Comment:
Spack doesn't know about any other versions of QT 3 besides `3.3.8b`, so these should be equivalent. I would imagine any other version of QT 3 would have the same requirement. What makes this change necessary?

Reply:
When I tried to build on my laptops spack didn't put the version number into the download url.  The 3.3.8b wasn't included, so the download of the tarball failed.

Reply:
HERE IS THE OUTPUT:

==> 20931: libmng: Successfully installed libmng
  Fetch: 0.03s.  Build: 7.42s.  Total: 7.46s.
[+] /home/jeg/ospack/spack/opt/spack/linux-fedora27-haswell/gcc-7.3.1/libmng-2.0.3-cv4mzvik5sdq2jtscsff5kbpm3sfp4ty
==> 20931: Installing qt
==> Warning: There is no checksum on file to fetch qt@3 safely.
==>   Fetch anyway? [y/N] y
==> Fetching with no checksum.
  Add a checksum or use --no-checksum to skip this check.
==> Fetching http://download.qt.io/archive/qt/3/qt-x11-free-3.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
==> Failed to fetch file from URL: http://download.qt.io/archive/qt/3/qt-x11-free-3.tar.gz
    URL http://download.qt.io/archive/qt/3/qt-x11-free-3.tar.gz was not found!
==> Fetching from http://download.qt.io/archive/qt/3/qt-x11-free-3.tar.gz failed.
==> Warning: Spack was unable to fetch url list due to a certificate verification problem. You can try running spack -k, which will not check SSL certificates. Use this at your own risk.

==> Cannot find version 3 in url_list
==> Error: FetchError: All fetchers failed for spack-stage-qt-3-26d6zsafjwylncljpris7irtvfvi3is4

/home/jeg/ospack/spack/lib/spack/spack/package.py:1127, in do_fetch:
       1124                raise FetchError("Will not fetch %s" %
       1125 self.spec.format('{name}{@version}'), ck_msg)
       1126
  >>   1127        self.stage.create()
       1128        self.stage.fetch(mirror_only)
       1129        self._fetch_time = time.time() - start_time
       1130


==> Error: Failed to install qt due to ChildError: FetchError: All fetchers failed for spack-stage-qt-3-26d6zsafjwylncljpris7irtvfvi3is4
/home/jeg/ospack/spack/lib/spack/spack/package.py:1127, in do_fetch:
       1124                raise FetchError("Will not fetch %s" %
       1125 self.spec.format('{name}{@version}'), ck_msg)
       1126
  >>   1127        self.stage.create()
       1128        self.stage.fetch(mirror_only)
       1129        self._fetch_time = time.time() - start_time
       1130

Traceback (most recent call last):
  File "/home/jeg/ospack/spack/lib/spack/spack/build_environment.py", line 801, in child_process
    return_value = function()
  File "/home/jeg/ospack/spack/lib/spack/spack/installer.py", line 1042, in build_process
    pkg.do_patch()
  File "/home/jeg/ospack/spack/lib/spack/spack/package.py", line 1167, in do_patch
    self.do_stage()
  File "/home/jeg/ospack/spack/lib/spack/spack/package.py", line 1152, in do_stage
    self.do_fetch(mirror_only)
  File "/home/jeg/ospack/spack/lib/spack/spack/package.py", line 1128, in do_fetch
    self.stage.fetch(mirror_only)
  File "/home/jeg/ospack/spack/lib/spack/spack/util/pattern.py", line 68, in getter
    getattr(item, self.name)(*args, **kwargs)
  File "/home/jeg/ospack/spack/lib/spack/spack/stage.py", line 476, in fetch
    raise fs.FetchError(err_msg, None)
spack.fetch_strategy.FetchError: All fetchers failed for spack-stage-qt-3-26d6zsafjwylncljpris7irtvfvi3is4

==> Warning: Skipping build of openspeedshop since qt failed
==> Error: Installation of openspeedshop failed.  Review log for details

Reply:
I think this is a similar issue to the discussion https://github.com/spack/spack/pull/14752#discussion_r382207238 (@scheibelp ). It was my mistake to interpret `qt@3` as being equivalent to`qt@3:3.999` Could you please change to `qt@3:3.9`? That should allow the concretization to pick up `3.3.8b` automatically while making it clearer that it just needs a QT3 version.

Reply:
That worked!  Thanks!   I pushed the change.